control-plane-operator/hostedclusterconfigoperator/controllers/nodecount/controller.go

```diff
@@ -0,0 +1,38 @@
+package nodecount
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	hypershiftv1beta1applyconfigurations "github.com/openshift/hypershift/client/applyconfiguration/hypershift/v1beta1"
+	hypershiftclient "github.com/openshift/hypershift/client/clientset/clientset"
+)
+
+type reconciler struct {
+	hcpName, hcpNamespace string
+	client                hypershiftclient.Interface
+
+	guestClusterClient client.Client
+}
+
+func (r *reconciler) Reconcile(ctx context.Context, _ reconcile.Request) (reconcile.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+	log.Info("Reconciling")
+
+	nodes := &corev1.NodeList{}
```

Comment:
I think we should return early if hcp is deleting or is paused.

Reply:
done